libs/designer/src/lib/core/state/workflow/workflowSlice.ts

```diff
@@ -248,6 +249,27 @@ export const workflowSlice = createSlice({
         }
         !!node?.children?.length && stack.push(...node.children);
       }
+
+      const positionChangesById = action.payload.nodes.reduce<Record<string, Node>>((acc, val) => {
+        if (!val.position) {
+          return acc;
+        }
+        return {
+          ...acc,
+          [val.id]: val,
+        };
+      }, {});
+      stack.push(state.graph);
+
+      while (stack.length) {
+        const node = stack.shift();
+        const change = positionChangesById[node?.id ?? ''];
+        const nodePosition = change?.position;
+        if (node && nodePosition) {
+          node.position = nodePosition;
+        }
+        !!node?.children?.length && stack.push(...node.children);
+      }
```

Comment:
With your current setup you're not just passing position changes, you're passing every node in the workflow.  This might become pretty impactful on performance on higher node count workflows if we update _every_ node's data on each change